### PR TITLE
JESI-2671

### PR DESCRIPTION
**Pull request by AndreTheHunter**
[JESI-2671](https://jesims.atlassian.net/browse/JESI-2671)
https://github.com/jesims/helios/pull/180

### CODE REVIEW

File: src/io/jesi/backpack/clojurescript.cljs [hunk 1]
```diff
@@ -19,16 +19,26 @@
...
-  "Transforms ClojureScript to JavaScript converting keys to camelCase"
-  (some->> x
-    (transform-keys ->camelCase)
-    clojure.core/clj->js))
+  "Transforms JavaScript to ClojureScript. Converting keys to kebab-case keywords by default"
+
+  ([js]
+   (js->clj js ->kebab-case-key))
+
+  ([js key-fn]
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/clojurescript.cljs]: Can now pass in `key-fn`

File: src/io/jesi/backpack/clojurescript.cljs [hunk 2]
```diff
@@ -19,16 +19,26 @@
...
+            clojure.core/js->clj
+            (transform-keys key-fn))))
+
+(defn clj->js
+  "Transforms ClojureScript to JavaScript. Converting keys to camelCase by default"
+
+  ([x]
+   (clj->js x ->camelCase))
+
+  ([o key-fn]
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/clojurescript.cljs]: Can now pass in `key-fn`

File: src/io/jesi/backpack/json.cljc [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
-    #?(:clj [io.jesi.backpack.string :refer [->kebab-case-key ->camelCase]])
+    #?(:clj [io.jesi.backpack.macros :refer [shorthand]])
     #?(:cljs [clojure.string :as string])
     #?(:cljs [io.jesi.backpack.clojurescript :refer [clj->js js->clj]])))
 
-(defn clj->json [x]
-  #?(:cljs (js/JSON.stringify (clj->js x))
-     :clj  (generate-string x {:key-fn ->camelCase})))
+;XXX DO NOT FORMAT THIS FILE
+;it breaks the function overloads
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/json.cljc]: ...stupid cursive

File: src/io/jesi/backpack/json.cljc [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
+  ([o key-fn]
+   #?(:cljs (js/JSON.stringify (clj->js o key-fn))
+      :clj  (generate-string o (shorthand key-fn)))))
+
+(defn json->clj
+
+  ([s]
+   (json->clj s ->kebab-case-key))
+
+  ([s key-fn]
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/json.cljc]: Can now pass in `key-fn`

File: src/io/jesi/backpack/json.cljc [hunk 5]
```diff
@@ -1,18 +1,32 @@
...
-             (some-> s
-                     js/JSON.parse
-                     js->clj))
-     :clj  (parse-string s ->kebab-case-key)))
+(defn clj->json
+
+  ([o]
+   (clj->json o ->camelCase))
+
+  ([o key-fn]
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/json.cljc]: Can now pass in `key-fn`

File: src/io/jesi/backpack/string.cljc [hunk 6]
```diff
@@ -39,14 +40,21 @@
   [s]
   (= s "true"))
 
+(defn- kw->str [k]
```
Comment (AndreTheHunter) [inline src/io/jesi/backpack/string.cljc]: Preserves keword namespace

File: src/io/jesi/backpack/fn.cljc [hunk 7]
```diff
@@ -40,3 +40,19 @@
...
 (def noop (constantly nil))
+
+(defn if-fn
+  "Higher-order if function.
+  Takes a predicate (`pred`), calling `then` or (optionally) `else` based on the predicate.
+  Returns nil if no `else` defined."
+
+  ([pred then]
+   (fn if-fn [v]
+     (if (pred v)
```
Comment (axrs) [inline src/io/jesi/backpack/fn.cljc]: `when` by convention

File: test/io/jesi/backpack/fn_test.cljc [hunk 8]
```diff
@@ -75,3 +75,26 @@
     (testing "derefs when not derefable"
       (assert-deref identity))))
 
+
+(deftest if-fn-test
+
+  (testing "if-fn"
+
+    (testing "is a function"
+      (is (ifn? bp/if-fn)))
```
Comment (axrs) [inline test/io/jesi/backpack/fn_test.cljc]: use `fn?`

**Pull request merged**